components/sup/src/manager/service/supervisor.rs

```diff
@@ -266,7 +266,7 @@ impl Supervisor {
                 // already allow broad reading of the PID file.
                 const PIDFILE_PERMISSIONS: u32 = 0o644;
                 let mut w = fs::AtomicWriter::new(pid_file)?;
-                w.with_permissions(PIDFILE_PERMISSIONS);
+                w.with_permissions(fs::Permissions::Explicit(PIDFILE_PERMISSIONS));
```

Comment:
If we wanted we could move the platform conditionality just around the definition of the pidfile permissions now that we have your nice abstraction.

Reply:
Good call! Addressed in https://github.com/habitat-sh/habitat/pull/7545/commits/4379e7d7b350fdf216d2670dd71a569b1ca42c40